grails-app/services/de/iteratec/osm/measurement/environment/wptserver/LocationPersisterService.groovy

```diff
@@ -54,7 +54,7 @@ class LocationPersisterService implements iLocationListener {
             List<Location> locations
             // for wpt versions above 2.18 there is a Browsers-Attribute
             List<String> browserNames = locationTagInXml.Browsers.size() != 0 ? locationTagInXml.Browsers.toString().split(",") : [locationTagInXml.Browser.toString()]
-            List<Browser> browsersForLocation = browserService.findAllByNameOrAlias(browserNames)
+            List<Browser> browsersForLocation = browserService.findAllByNameOrCreate(browserNames, false)
             browsersForLocation.each { currentBrowser ->
                 String uniqueIdentfierForServer = locationTagInXml.id.toString().endsWith(":${currentBrowser.name}") ?: locationTagInXml.id.toString() + ":${currentBrowser.name}"
```

Comment:
Please also fix the creation of the identifier by using only data from the XML and *not* the entities name.
If the WPT server location FooLocation provides a browser called `IE` which is configured in OSM as an alias for a browser with the name `InternetExplorer`, this logic will create an identifier called `FooLocation:InternetExplorer`.
However, the WPT server doesn't understand this identifier, as it doesn't know our aliases. So the identifier needs to be `FooLocation:IE`:
To achieve this, you could iterate over `browserNames` and call `findOrCreateByNameOrAlias` for one *single* browser. So you have both access to the original name and the associated entity.

Maybe you could extract the identifier-creation logic into a new function that can be unit-tested.